sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/RxGatewayStoreModel.java

```diff
@@ -456,13 +456,13 @@ private void applySessionToken(RxDocumentServiceRequest request) {
                         !Strings.areEqual(requestConsistencyLevel, ConsistencyLevel.EVENTUAL.toString())));
 
         if (!Strings.isNullOrEmpty(request.getHeaders().get(HttpConstants.HttpHeaders.SESSION_TOKEN))) {
-            if (!sessionTokenApplicable || ReplicatedResourceClientUtils.isMasterResource(request.getResourceType())) {
+            if (!sessionTokenApplicable || isMasterOperation(request.getResourceType(), request.getOperationType())) {
                 request.getHeaders().remove(HttpConstants.HttpHeaders.SESSION_TOKEN);
             }
             return; //User is explicitly controlling the session.
         }
 
-        if (!sessionTokenApplicable || ReplicatedResourceClientUtils.isMasterResource(request.getResourceType())) {
+        if (!sessionTokenApplicable || isMasterOperation(request.getResourceType(), request.getOperationType())) {
```

Comment:
The apply session token today is a little bit different today across Java v2, v4 and .net SDK.

1. sessionTokenApplicable:  In java V4: we also skip applying the session token when Eventual Consistency is explicitly requested on request-level for data plane operations. This is not included in V2 and .net SDK.


2. If user pass the session token, how we handle it: I think in V4, we will only keep it for session, but in V2 and .net SDK, we may keep it for non-session consistency. 

After this change, what will happen in Java V4:
1. Trigger and UDF will be treated as master resources, and same as other master resource, we do not capture session token
2. We will not apply session token for query plan, trigger, UDF, and execute stored procedure

Reply:
`sessionTokenApplicable: In java V4: we also skip applying the session token when Eventual Consistency is explicitly requested on request-level for data plane operations. This is not included in V2 and .net SDK.`

If consistency is eventual there is no need to set session token. This is intentional. 

Does this PR change this behaviour?

Reply:
@moderakh No, I did not change it. Just point out that V2 and .Net SDK does not have that specific check. We added the eventual check in this PR: https://github.com/Azure/azure-sdk-for-java/pull/16111/files

Reply:
we don't include session token for eventual consistency on purpose the intention is to bypass using SessionContainer. Not using that helps to avoid unnecessary perf hit. V2 and DotNet SDK as you said do not have this optimization.

we gain 5-10% perf improvement by not accessing SessionContainer when not needed.
see this PR for more info when we introduced this optimization: https://github.com/Azure/azure-sdk-for-java/pull/8064

Reply:
wow I see, thanks @moderakh